torch/_inductor/lowering.py

```diff
@@ -1126,10 +1122,33 @@ def can_fuse_reduction(t):
 
     # TODO: We observed negative performance impact of pointwise_cat optimization on CPU so disabled it.
     #             We will revisit this later after enabling vectorization on index_expr.
-    if (
-        len(inputs) <= config.max_pointwise_cat_inputs
-        and inputs[0].get_device().type != "cpu"
-        and not fusable_reduction
+    if inputs[0].get_device().type == "cpu" or fusable_reduction:
+        return TensorBox(ir.ConcatKernel.create(inputs, dim))
+
+    def op_count(x):
+        if isinstance(x, (TensorBox, ir.StorageBox)):
+            return op_count(unwrap_tensor(x))
+
+        # this will correspond to a direct memory read
+        if not isinstance(x, ir.Pointwise):
+            return 0
```

Comment:
Q: what if x is an reduction. i.e., we cat a few reductions together. We would under-estimate the op_count here.  (A bit picky since this may not appear in real model much)

Reply:
In prior PR I disable inputs with reduction reads